cob_obstacle_distance/src/parsers/stl_parser.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2014 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_obstacle_distance
+ *
+ * \author
+ *   Author: Marco Bezzon, email: Marco.Bezzon@ipa.fraunhofer.de
+ *
+ * \date Date of creation: July, 2015
+ *
+ * \brief
+ *   STL file parser.
+ *   Source from http://www.sgh1.net/b4/cpp-read-stl-file
+ *
+ */
+#include <boost/filesystem.hpp>
+#include <ros/ros.h>
+#include <fstream>
+
+#include "cob_obstacle_distance/parsers/stl_parser.hpp"
+#include "cob_obstacle_distance/helpers/helper_functions.hpp"
+
+/**
+ * Hard coded STL file parsing according to binary file specification in https://en.wikipedia.org/wiki/STL_%28file_format%29.
+ */
+int8_t StlParser::read(std::vector<TriangleSupport>& tri_vec)
```

Comment:
Yes correct. For further developments this can be used as example implementation. Currently not used -> no code from compiler generated -> no overhead -> win-win